beacon-chain/state/fieldtrie/field_trie_helpers.go

```diff
@@ -148,34 +138,6 @@ func convertBalances(indices []uint64, elements interface{}, convertAll bool) ([
 	return handleBalanceSlice(val, indices, convertAll)
 }
 
-// handleByteArrays computes and returns byte arrays in a slice of root format.
-func handleByteArrays(val [][]byte, indices []uint64, convertAll bool) ([][32]byte, error) {
```

Comment:
The behavior of this function is completely different if `convertAll`  is set or not. I would avoid completely the boolean, but even if you want to keep it, it's cleaner to have three different helpers
```
handleByteArrays(val, indices, convertAll) {
         if convertAll {
            return handleAll (val)
          }
          return handleSome(val, indices)
      }
  func handleAll(val)
  func handleSome(val, indices)
  ```

Reply:
I am removing this function all together. If this comment is about `handle32ByteArrays`, then I'm happy to change in a later PR because this one didn't touch the function